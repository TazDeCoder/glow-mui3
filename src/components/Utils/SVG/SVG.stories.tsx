import type { ComponentMeta, ComponentStory } from '@storybook/react';

import { SVG } from './SVG';

export default {
  title: 'Components/Utils/SVG',
  component: SVG,
} as ComponentMeta<typeof SVG>;

export const Path: ComponentStory<typeof SVG> = () => (
  <SVG width={400} height={210}>
    <path d="M150 0 L75 200 L225 200 Z" />
  </SVG>
);
