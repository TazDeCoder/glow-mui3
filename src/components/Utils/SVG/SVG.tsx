/** @jsxRuntime classic */
/** @jsx jsx */

import { jsx } from '@theme-ui/core';
import type { SVGAttributes } from 'react';

import { Box, BoxOwnProps } from '../Box';
import type { Assign } from '../../types';

export interface SVGProps
  extends Assign<SVGAttributes<SVGElement>, BoxOwnProps> {
  /** Size of the component (width/height)
   * @default 24
   */
  size?: number | string;
}

export const SVG: React.FC<SVGProps> = ({
  children,
  size = 24,
  ...props
}: SVGProps) => {
  const svgProps: SVGProps = {
    xmlns: 'http://www.w3.org/2000/svg',
    width: size,
    height: size,
    fill: 'currentColor',
    ...props,
  };

  return (
    <Box as="svg" {...(svgProps as {})}>
      {children}
      Sorry, your browser does not support inline SVG
    </Box>
  );
};
