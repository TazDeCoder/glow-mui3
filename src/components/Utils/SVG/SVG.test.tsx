import { render } from '@testing-library/react';

import { SVG } from './SVG';

describe('<SVG />', () => {
  describe('Snapshots', () => {
    it('renders correctly', () => {
      const { container } = render(<SVG>Foo</SVG>);
      expect(container).toMatchSnapshot();
    });
  });
});
