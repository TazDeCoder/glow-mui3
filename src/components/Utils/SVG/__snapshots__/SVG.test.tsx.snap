// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SVG /> Snapshots renders correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  width: 24px;
  height: 24px;
}

<div>
  <svg
    class="emotion-0"
    fill="currentColor"
    xmlns="http://www.w3.org/2000/svg"
  >
    Foo
    Sorry, your browser does not support inline SVG
  </svg>
</div>
`;
